components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -1511,6 +1511,13 @@ private static boolean isNotActiveState(String appState) {
         return !APP_STATE_ACTIVE.equalsIgnoreCase(appState);
     }
 
+    /**
+     * This method retrieves the service provider tenant domain using the client ID.
+     * However, internally it uses the tenant present in the carbon context.
```

Comment:
if it is internally using the tenant present in the carbon context, can't we directly return that tenant instead of again trying to derive it from the service provider?

Reply:
Then we'll loose the application validation part. If an application not found or an error occurs, this method returns the super tenant domain. Tenant in the context could be a different one. Hence it will be a behavior change